apps/workers/email-composer/src/filterRoles.ts

```diff
@@ -0,0 +1,20 @@
+import { createClient } from 'redis'
+import { RedisPrefix } from 'shared/src/enums/redis'
+
+export const filterRoles = async (rolesId: string[]) => {
+  let concatenatedHtmlRoles = ''
+  const redisClient = createClient()
+  await redisClient.connect()
+
+  for (let index = 0; index < rolesId.length; index++) {
+    const roleId = rolesId[index]
+
+    const roleHtml = await redisClient.get(
+      `${RedisPrefix.RolesRenderer}${roleId}`
+    )
+    if (!roleHtml) break
+    concatenatedHtmlRoles = `${concatenatedHtmlRoles}${roleHtml}`
+  }
```

Comment:
This code is problematic.

- The for loop won't wait for the `await`, if we need to await we should use the `for`...`of`
- You're only using `index` to access the item on the list, so you don't really need a `for`. Replace with forEach, map or for...of